braid/braid.h

```diff
@@ -190,19 +201,35 @@ typedef braid_Int
                           braid_Real    *norm_ptr  /**< output, norm of braid_Vector (this is a spatial norm) */ 
                           );
 
+/**
+ * (optional) Compute an inner (scalar) product between two braid_Vectors
+ * *prod_ptr* = <*u*, *v*>
+ * Only needed when using Delta correction
+ * 
+ * The most common choice would be the standard dot product.
+ * Vectors are normalized under the norm induced by this inner product,
+ * *not* the function defined in SpatialNorm, which is only used for halting
+ */
+typedef braid_Int
+(*braid_PtFcnInnerProd)(braid_App        app,       /**< user-defined _braid_App structure */
+                        braid_Vector     u,         /**< first vector */
+                        braid_Vector     v,         /**< second vector */
+                        braid_Real      *prod_ptr   /**< output, result of inner product */
+                        );
+
 /**
  * Gives user access to XBraid and to the current vector *u* at time *t*.  Most
  * commonly, this lets the user write the vector to screen, file, etc...  The
  * user decides what is appropriate.  Note how you are told the time value *t*
  * of the vector *u* and other information in *status*.  This lets you tailor
  * the output, e.g., for only certain time values at certain XBraid iterations.
- * Querrying status for such information is done through
+ * Querying status for such information is done through
```

Comment:
Thank you for catching all these typos!

Reply:
I have an extension in my IDE that highlights them in red for me, so I can't miss them :)